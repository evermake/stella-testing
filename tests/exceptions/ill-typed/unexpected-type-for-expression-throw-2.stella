language core;

extend with #exceptions, #exception-type-declaration;

exception type = Bool

fn main(n : Nat) -> Bool {
  return throw(succ(0))
}

