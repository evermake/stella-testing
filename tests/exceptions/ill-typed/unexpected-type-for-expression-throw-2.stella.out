An error occurred during typechecking!
===================
Primary type error:
===================
Type Error Tag: [ERROR_UNEXPECTED_TYPE_FOR_EXPRESSION]
expected type
  Bool
but got
  Nat
for expression
  succ (0)
========================
Alternative type errors:
========================
Type Error Tag: [ERROR_AMBIGUOUS_THROW_TYPE]
cannot infer type for throw